aviary/utils/preprocessors.py

```diff
@@ -35,142 +35,201 @@ def preprocess_crewpayload(aviary_options: AviaryValues):
+    mission_sum = 0
+    design_sum = 0
+    for i in range(1, len(pax_keys)):
+        mission_sum += aviary_options.get_val(pax_keys[i])
+        design_sum += aviary_options.get_val(design_pax_keys[i])
+
+    # Resolve conflicts between seat class totals and num_passengers
+    # Specific beats general - trust the individual class counts over the total provided, unless
+    #    the class counts are all zero, in which case trust the total provided and assume all economy
+    # design num_passengers does not match sum of seat classes for design
+    if design_pax != design_sum:
+        # if sum of seat classes is zero (design pax is not), assume all economy
+        if design_sum == 0:
+            if verbosity >= 1:
+                UserWarning(
+                    "Information on seat class distribution for aircraft design was "
+                    "not provided - assuming that all passengers are economy class."
+                )
+            aviary_options.set_val(
+                Aircraft.CrewPayload.Design.NUM_TOURIST_CLASS, design_pax
+            )
+        else:
+            if verbosity >= 1:
+                UserWarning(
+                    "Sum of all passenger classes does not equal total number of "
+                    "passengers provided for aircraft design. Overriding "
+                    "Aircraft.CrewPayload.Design.NUM_PASSENGERS with the sum of "
+                    "passenger classes for design."
+                )
+            aviary_options.set_val(
+                Aircraft.CrewPayload.Design.NUM_PASSENGERS, design_sum
+            )
+
+    # mission num_passengers does not match sum of seat classes for mission
+    if mission_pax != mission_sum:
+        # if sum of seat classes is zero (mission pax is not), assume all economy
+        if mission_sum == 0:
+            if verbosity >= 1:
+                UserWarning(
+                    "Information on seat class distribution for current mission was "
+                    "not provided - assuming that all passengers are economy class."
+                )
+            aviary_options.set_val(Aircraft.CrewPayload.NUM_TOURIST_CLASS, mission_pax)
+        else:
+            if verbosity >= 1:
+                UserWarning(
+                    "Sum of all passenger classes does not equal total number of "
+                    "passengers provided for current mission. Overriding "
+                    "Aircraft.CrewPayload.NUM_PASSENGERS with the sum of "
+                    "passenger classes for the flown mission."
+                )
+            aviary_options.set_val(Aircraft.CrewPayload.NUM_PASSENGERS, mission_sum)
+
+    # Check cases where mission passengers are greater than design passengers
+    design_pax = aviary_options.get_val(Aircraft.CrewPayload.Design.NUM_PASSENGERS)
+    mission_pax = aviary_options.get_val(Aircraft.CrewPayload.NUM_PASSENGERS)
+    if mission_pax > design_pax:
+        if verbosity >= 1:
+            # TODO Should this be an error?
```

Comment:
yeah, i would just warn on this, just in case the user wants to run a mission overloaded with passengers (though where do they sit?)

Reply:
https://www.guinnessworldrecords.com/world-records/most-passengers-on-an-aircraft
Over 1k people were crammed on a 747 once! So we'd be able to model that flight now